### PR TITLE
Add timezone to dags in main screen

### PR DESCRIPTION
**Pull request by miky-schreiber**
---
Adds the timezone of the DAG to the cron expression in Airflow's main screen
It looks like this:

image

closes: [#27949](https://github.com/apache/airflow/discussions/27949)

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pierrejeambrun — DISMISSED** [review]

**Comment by pierrejeambrun** [comment]
Is it revelent to always show the timezone, even for `Dataset` triggered Dag and not scheduled one. (`None`).

It looks like this for instance:
image

**Comment by pierrejeambrun** [comment]
Is it revelent to always show the timezone, even for `Dataset` triggered and `not scheduled` dags ?

It looks like this with these change:
image

**Review by uranusjr — CHANGES_REQUESTED** [review]
I’d change this to something like <a ...>`{{ dag.schedule.description }}</a>` and implement that property instead. This display does not make sense except for cron expressions.

**Comment by miky-schreiber** [comment]
> Is it relevent to always show the timezone, even for `Dataset` triggered and `not scheduled` dags ?

@pierrejeambrun I modified the code to add timezone only when the schedule is a cron expression, would appreciate your feedback.
@uranusjr I changed it to schedule_description as you recommended, thank you for this wise tip.

File: airflow/models/dag.py [hunk 1]
```diff
@@ -3204,6 +3204,13 @@ def __init__(self, concurrency=None, **kwargs):
     def __repr__(self):
         return f"<DAG: {self.dag_id}>"
 
+    @property
+    def schedule_description(self) -> str | None:
```
Comment (pierrejeambrun) [inline airflow/models/dag.py:3223]: This is not enough, dataset triggered dag also have a schedule_interval as a string, I think we need to check the self.timetable attribute.
image

Comment (pierrejeambrun) [inline airflow/models/dag.py:3223]: Type is wrong here I believe:
```suggestion
    def schedule_description(self) -> str:
```

**Review by pierrejeambrun — DISMISSED** [review]
Thanks for the update @miky-schreiber

A test is failing. Properties are not serialized I believe (which makes sense as they are derived from other attributes and not part of the `__dict__` method).

Dataset schedule need to be handled cf below comment.

File: airflow/models/dag.py (hunk 1 above)
Comment (miky-schreiber) [reply airflow/models/dag.py:3223]: @pierrejeambrun the DagModel class doesn't have self.timetable attribute. I used self.timetable_description instead.

**Comment by eladkal** [comment]
I think we should avoid presenting timezone if it's the same as the default timezone.
For example users who use only UTC should not see UTC every-time, they know everything is UTC.

File: airflow/models/dag.py [hunk 2]
```diff
@@ -3204,6 +3204,17 @@ def __init__(self, concurrency=None, **kwargs):
...
+        if (
+            isinstance(self.schedule_interval, str)
+            and self.schedule_interval != "@once"
+            and self.timetable_description != DatasetTriggeredTimetable.description
+        ):
+            return str(self.schedule_interval) + " " + str(self.timezone.name)
```
Comment (uranusjr) [inline airflow/models/dag.py:3229]: This weird isinstance check makes me feel the entire logic should go into the timetable class instead. A timetable can decide whether it wants to take `timezone`, and we can change `description` to include the timezone information.

(Note that `schedule_interval` should _not_ include timezone information for backward compatibility considerations, but we can change the display of `timetable_description` so an additional property is not needed.

**Review by pierrejeambrun — COMMENTED** [review]
unblock

**Pull request closed without merging**